server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryProperties.java

```diff
@@ -0,0 +1,76 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailrepository.api;
+
+import java.util.Objects;
+
+import com.google.common.base.MoreObjects;
+
+public class MailRepositoryProperties {
+
+    public static class Builder {
+        private boolean browsable;
+
+        public Builder browsable(boolean browsable) {
+            this.browsable = browsable;
+            return this;
```

Comment:
I would have a returned an immutable copy of the builder.
What is the team opinion on this subject ?

Reply:
> I would have returned an immutable copy of the builder

IDK how to do it :). 

BTW the reason I wrote that Builder is: I think It would be helpful in the CassandraMailRepositoryPropertiesStore ticket, to make Jackson can use the proper builder (the traditional builder) when deserializing json string to POJO. Maybe it's over anticipated. I will update the code to not use that builder.

Reply:
You just create a new instance of the builder with the new value. 
As for now only browsable allow to change a property of the builder, the builder will then be immutable.